spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialAveragePooling.scala

```diff
@@ -186,6 +200,24 @@ class SpatialAveragePooling[@specialized(Float, Double) T: ClassTag](
     }
   }
 
+  // return (padTop, padDown, padLeft, padRight)
+  protected def getPadding(inputHeight: Int, inputWidth: Int): (Int, Int, Int, Int) = {
+    if (padW == -1 && padH == -1) {
+      // deal with SAME padding
+      val oW = Math.ceil(inputWidth.toFloat / dW.toFloat).toInt
```

Comment:
It is correct actually. In the "SAME" padding algorithm, when padding is need, the algorithm will guarantee `inputHeight + padAlongH - kH` is always divisible by `dH` where the notion of `ceiling` and `floor` does not make sense. However, the code is a little confusing, I'll try to refactor that.